tests/naive.py

```diff
@@ -950,14 +974,25 @@ def update(self, t):
         else:
             self._T_isfinite[-1] = False
             self._T[-1] = 0
+
+        self._T_subseq_isconstant[:] = np.roll(self._T_subseq_isconstant, -1)
+        self._T_subseq_isconstant[-1] = rolling_isconstant(
+            self._T[-self._m :], self._m, self._T_subseq_isconstant_func
+        ) & np.all(self._T_isfinite[-self._m :])
+
         self._n_appended += 1
 
         self._P = np.roll(self._P, -1, axis=0)
         self._I = np.roll(self._I, -1, axis=0)
         self._left_P[:] = np.roll(self._left_P, -1)
         self._left_I[:] = np.roll(self._left_I, -1)
 
-        D = core.mass(self._T[-self._m :], self._T)
+        D = core.mass(
+            self._T[-self._m :],
+            self._T,
+            T_subseq_isconstant=self._T_subseq_isconstant,
+            Q_subseq_isconstant=self._T_subseq_isconstant[[-1]],
```

Comment:
I am using [-1] as index so that it returns a numpy array rather than just a boolean value. 

Alternative approach: we might add support for `bool`. In other words, if I pass a bool, `core.rolling_isconstant`  converts it to an array of size 1 internally.

Reply:
This is probably fine. Thanks for asking